### PR TITLE
Correctly weight clusters in ATE and APE functions.

### PR DESCRIPTION
**Pull request by swager**
When clusters are specified in grf, the default behavior is to give each cluster (not each sample) equal weight in training. However, until now, the ATE and APE functions did not respect this convention, and instead gave each sample equal weight. This PR fixes the issues, and brings the ATE and APE functions in line with the rest of the package.

### CODE REVIEW

File: r-package/grf/R/average_partial_effect.R [hunk 1]
```diff
@@ -91,13 +111,13 @@ average_partial_effect = function(forest, calibrate.weights = TRUE, subset = NUL
   
   # Estimate variance using the calibration
   if (length(subset.clusters) == 0) {
-    cape.se = sqrt(mean((debiasing.weights * plugin.residual)^2) / length(subset.W.orig))
+    cape.se = sqrt(mean((debiasing.weights * plugin.residual)^2) / (length(subset.W.orig) - 1))
```
Comment (jtibshirani) [inline r-package/grf/R/average_partial_effect.R]: I'm curious why we've added this `- 1` here?

File: r-package/grf/R/forest_summary.R [hunk 2]
```diff
@@ -28,30 +28,47 @@
 #'
 #' @export
 test_calibration = function(forest) {
```
Comment (jtibshirani) [inline r-package/grf/R/forest_summary.R]: Should we make a note here about clusters, as we did for the other methods?

File: r-package/grf/R/average_partial_effect.R [hunk 3]
```diff
@@ -13,6 +18,7 @@
 #'               estimate the ATE. WARNING: For valid statistical performance,
 #'               the subset should be defined only using features Xi, not using
 #'               the treatment Wi or the outcome Yi.
+#' @param num.trees.for.variance Number of trees used to estimate Var[Wi | Xi = x].
```
Comment (jtibshirani) [inline r-package/grf/R/average_partial_effect.R]: It's generally good to make unrelated changes like this in a separate PR. At the least, maybe we could add a note to the PR description noting these additional changes?

File: r-package/grf/R/average_partial_effect.R (hunk 1 above)
Comment (swager) [reply r-package/grf/R/average_partial_effect.R]: It's by analogy to <a href="https://en.wikipedia.org/wiki/Bessel%27s_correction">Bessel's correction</a> for variance estimation. Doesn't really make a difference in large samples, but still the right thing to do.

File: r-package/grf/R/forest_summary.R (hunk 2 above)
Comment (swager) [reply r-package/grf/R/forest_summary.R]: I think we also need a bigger PR where we explain how to use the `test_calibration` function. Maybe we can work on that discussion then? The way we handle clusters here is not consistent with the rest of the package (i.e., 1 cluster = 1 vote).
